### PR TITLE
Add MessagePack::Any convenience wrapper

### PR DESCRIPTION
**Pull request by willhbr**
Like JSON::Any and YAML::Any, allows for deserialising data and accessing values in a dynamic fashion without too much boilerplate.

### CODE REVIEW

File: src/message_pack/any.cr [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        @raw.is_a? {{ type }}
+      end
+    {% end %}
+
+    {% for name, type in {
+                           i64:  Int64,
+                           i32:  Int32,
+                           i16:  Int32,
+                           i8:   Int32,
+                           ui64: UInt64,
```
Comment (kostya) [inline src/message_pack/any.cr]: use just `u` better here, i think (u64)

File: src/message_pack/any.cr [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    end
+
+    {% for name, type in {
+                           nil:    Nil,
+                           bool:   Bool,
+                           float:  Float64,
+                           string: String,
+                           hash:   Hash(Type, Type),
+                           array:  Array(Type),
+                           bytes:  Bytes,
```
Comment (kostya) [inline src/message_pack/any.cr]: since Bytes not merged, would it work?

File: spec/any_spec.cr [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  it "wraps array" do
+    serialized = [1, 2, "three", false].to_msgpack
+    any = MessagePack::Any.from_msgpack IO::Memory.new(serialized)
+    any[0].as_i32.should eq(1)
+    any[2].as_string.should eq("three")
+    any[100]?.should eq(nil)
+    any.size.should eq(4)
+  end
```
Comment (kostya) [inline spec/any_spec.cr:156]: any spec for methods `is_` and `each`

File: spec/any_spec.cr [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+end
```
Comment (kostya) [inline spec/any_spec.cr:157]: can you copy more specs from json any spec?

File: src/message_pack/any.cr (hunk 1 above)
Comment (willhbr) [reply src/message_pack/any.cr]: Done

File: src/message_pack/any.cr (hunk 2 above)
Comment (willhbr) [reply src/message_pack/any.cr]: Removed, it worked because nothing was calling it

File: spec/any_spec.cr (hunk 3 above)
Comment (willhbr) [reply spec/any_spec.cr:156]: Copied specs from JSON::Any

**Comment by willhbr** [comment]
Thanks!

**Pull request merged**